pkg/meta/sql.go

```diff
@@ -3900,3 +3907,60 @@ func (m *dbMeta) doAttachDirNode(ctx Context, parent Ino, inode Ino, name string
 		return err
 	}, parent))
 }
+
+func (m *dbMeta) touchAtime(ctx Context, ino Ino, cur *Attr) (rerr syscall.Errno) {
+	if (m.conf.AtimeMode != StrictAtime && m.conf.AtimeMode != RelAtime) || m.conf.ReadOnly {
+		return 0
+	}
+
+	var curNode = node{Inode: ino}
+	var attr *Attr
+	newAttr := &Attr{}
+	if cur != nil {
+		attr = cur
+	} else if m.of.Check(ino, newAttr) {
+		attr = newAttr
+	}
+
+	defer func() {
+		if rerr == 0 && m.of.IsOpen(ino) {
+			m.of.Update(ino, attr)
+		}
+	}()
+
+	now := time.Now()
+
+	// Already got attr, update atime without transaction
+	if attr != nil {
+		if !m.atimeNeedsUpdate(attr, now) {
+			return 0
+		}
+		curNode.Atime = now.Unix()*1e6 + int64(now.Nanosecond())/1e3
+		s := m.db.NewSession()
+		defer s.Close()
+		_, err := s.Cols("atime").Update(&curNode, &node{Inode: ino})
+		return errno(err)
+	}
+
+	// Should get attr first, do it in transaction
+	return errno(m.txn(func(s *xorm.Session) error {
+		attr = newAttr
+		if !m.of.Check(ino, attr) {
```

Comment:
no need to check again, just like that in Redis

Reply:
Fixed